docs/ldap.md

```diff
@@ -62,17 +71,18 @@ authentication source:
         'search.enable' => false,
 
         /*
-         * The DN which will be used as a base for the search.
-         * This can be a single string, in which case only that DN is searched, or an
-         * array of strings, in which case they will be searched in the order given.
+         * An array on DNs which will be used as a base for the search.
+         * In case of multiple strings, they will be searched in the order given.
          */
-        'search.base' => 'ou=people,dc=example,dc=org',
+        'search.base' => [
+            'ou=people,dc=example,dc=org',
+        ],
```

Comment:
That's actually something I wanted to discuss.. I went with this so I didn't have to use `Configuration::getArrayize`.. Static analysis tools are having a hard time to detect the return-type for it..  The other getter-methods on `Configuration` also have the 'flaw' that the `default` parameter is of mixed type, so we can never really know whether i.e. `getString` will actually return a string or not..  Ideally I'd change this in SSP `master` so the default value is either null or the expected type.. What do you think? For the possible edge-cases someone can still use `getValue` instead.1

As far is the BC-break goes... I don't mind changing it back to what it was..

Reply:
Yes, but I see that also quite some other options have changed, sometimes also their structure or their value. This may be acceptable to jump straight away to a "good" configuration instead of carrying more legacy. Only thing is that no-one ever got warned about deprecation. But we might just accept it. And document clearly that this changed and recommend to ... use a fresh template and copy your settings or something like that?